app/main.py

```diff
@@ -1,6 +1,26 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached = []
+    arg_cache = []
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        nonlocal cached, arg_cache
+
+        if args not in arg_cache:
+            print("Calculating new result")
+            result = func(*args)
+            cached.append(result)
+            arg_cache.append(args)
+
+            return result
+
+        else:
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.